lib/src/test/java/uk/gov/di/ipv/core/library/service/CriResponseServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.di.ipv.core.library.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.di.ipv.core.library.persistence.DataStore;
+import uk.gov.di.ipv.core.library.persistence.item.CriResponseItem;
+
+import java.time.Instant;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static uk.gov.di.ipv.core.library.fixtures.TestFixtures.SIGNED_VC_1;
+
+@ExtendWith(MockitoExtension.class)
+public class CriResponseServiceTest {
+    @Mock private ConfigService mockConfigService;
+
+    @Mock private DataStore<CriResponseItem> mockDataStore;
+
+    private CriResponseService criResponseService;
+
+    private static final String USER_ID_1 = "user-id-1";
+
+    @BeforeEach
+    void setUp() {
+        criResponseService = new CriResponseService(mockConfigService, mockDataStore);
+    }
+
+    @Test
+    void shouldReturnCredentialFromDataStoreForSpecificCri() {
+        String ipvSessionId = "ipvSessionId";
+        String criId = "criId";
+        CriResponseItem criResponseItem =
+                createCriResponseStoreItem(USER_ID_1, "ukPassport", SIGNED_VC_1, Instant.now());
+
+        when(mockDataStore.getItem(ipvSessionId, criId)).thenReturn(criResponseItem);
+
+        CriResponseItem retrievedCredentialItem =
```

Comment:
Just minor naming niggles - maybe `retrievedCriResponseItem` ?

And the test labels here and below could be 'shouldReturnCriResponseItem' rather than '...Credential'